include/tscpp/api/GlobalPlugin.h

```diff
@@ -82,7 +83,15 @@ class GlobalPlugin : public Plugin
   GlobalPlugin(bool ignore_internal_transactions = false);
 
 private:
-  GlobalPluginState *state_; /**< Internal state tied to a GlobalPlugin */
+  struct PluginState;
+  PluginState *state_; /**< Internal state tied to a GlobalPlugin */
 };
 
+bool RegisterGlobalPlugin(const char *name, const char *vendor, const char *email);
+inline bool
+RegisterGlobalPlugin(std::string const &name, std::string const &vendor, std::string const &email)
+{
+  return RegisterGlobalPlugin(name.c_str(), vendor.c_str(), email.c_str());
```

Comment:
Shouldn't these be either `std::string const&` or `std::string_view`? Why bother with `c_str()`?

Reply:
You mean eliminate the first overload of RegisterGlobalPlugin()?